cpp/src/arrow/compute/api_vector.cc

```diff
@@ -111,6 +111,10 @@ static auto kSortOptionsType =
     GetFunctionOptionsType<SortOptions>(DataMember("sort_keys", &SortOptions::sort_keys));
 static auto kPartitionNthOptionsType = GetFunctionOptionsType<PartitionNthOptions>(
     DataMember("pivot", &PartitionNthOptions::pivot));
+static auto kSelectKOptionsType = GetFunctionOptionsType<SelectKOptions>(
+    DataMember("k", &SelectKOptions::k), DataMember("keys", &SelectKOptions::keys),
+    DataMember("order", &SelectKOptions::order),
+    DataMember("keep", &SelectKOptions::keep));
```

Comment:
nit: it'd be good to add a couple instances of the options to the test in function_test.cc: https://github.com/apache/arrow/blob/3da09e51e3f94ca427915c98ac59d72701f728bc/cpp/src/arrow/compute/function_test.cc#L112